connection.go

```diff
@@ -49,7 +52,19 @@ func NewConnection(address string, opts *ConnectOpts) (*Connection, error) {
 	}
 	// Connect to Server
 	nd := net.Dialer{Timeout: c.opts.Timeout}
-	c.conn, err = nd.Dial("tcp", address)
+	if !c.opts.SSL {
+		c.conn, err = nd.Dial("tcp", address)
+	} else {
+		roots := x509.NewCertPool()
+		ok := roots.AppendCertsFromPEM([]byte(c.opts.CaCert))
+		if !ok {
+			panic("failed to parse root certificate")
```

Comment:
I think it would make more sense for an error to be returned here instead of panicking, what do you think?